common/src/main/java/me/lucko/luckperms/common/plugin/AbstractLuckPermsMod.java

```diff
@@ -0,0 +1,370 @@
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ */
+
+package me.lucko.luckperms.common.plugin;
+
+import me.lucko.luckperms.common.actionlog.LogDispatcher;
+import me.lucko.luckperms.common.api.ApiRegistrationUtil;
+import me.lucko.luckperms.common.api.LuckPermsApiProvider;
+import me.lucko.luckperms.common.calculator.CalculatorFactory;
+import me.lucko.luckperms.common.config.AbstractConfiguration;
+import me.lucko.luckperms.common.config.ConfigKeys;
+import me.lucko.luckperms.common.config.LuckPermsConfiguration;
+import me.lucko.luckperms.common.config.adapter.ConfigurationAdapter;
+import me.lucko.luckperms.common.context.LPStaticContextsCalculator;
+import me.lucko.luckperms.common.dependencies.Dependency;
+import me.lucko.luckperms.common.dependencies.DependencyManager;
+import me.lucko.luckperms.common.event.AbstractEventBus;
+import me.lucko.luckperms.common.event.EventDispatcher;
+import me.lucko.luckperms.common.extension.SimpleExtensionManager;
+import me.lucko.luckperms.common.inheritance.InheritanceHandler;
+import me.lucko.luckperms.common.locale.LocaleManager;
+import me.lucko.luckperms.common.locale.message.Message;
+import me.lucko.luckperms.common.messaging.InternalMessagingService;
+import me.lucko.luckperms.common.messaging.MessagingFactory;
+import me.lucko.luckperms.common.plugin.logging.PluginLogger;
+import me.lucko.luckperms.common.sender.Sender;
+import me.lucko.luckperms.common.storage.Storage;
+import me.lucko.luckperms.common.storage.StorageFactory;
+import me.lucko.luckperms.common.storage.StorageType;
+import me.lucko.luckperms.common.storage.implementation.file.watcher.FileWatcher;
+import me.lucko.luckperms.common.tasks.SyncTask;
+import me.lucko.luckperms.common.treeview.PermissionRegistry;
+import me.lucko.luckperms.common.verbose.VerboseHandler;
+import me.lucko.luckperms.common.web.BytebinClient;
+import net.luckperms.api.LuckPerms;
+import okhttp3.OkHttpClient;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.EnumSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Represents an implementation of a LuckPermsPlugin for a modded context where the plugin must be loaded very early on.
+ */
+public abstract class AbstractLuckPermsMod implements LuckPermsPlugin {
```

Comment:
This seems to be functionally identical to `AbstractLuckPermsPlugin`, only with the addition of `initializeMod()` - could this method not just be added to the other class? Seems silly to duplicate all of this.